src/methods/properties.js

```diff
@@ -122,7 +122,14 @@ function InternalUpdatedProperty(realm: Realm, O: ObjectValue, P: PropertyKeyVal
     } else {
       generator.emitPropertyDelete(O, P);
     }
-  } else if (!desc.configurable && desc.enumerable && O === realm.$GlobalObject && desc.value !== undefined) {
+  } else if (
+    flag === "propertyCreation" &&
```

Comment:
With this change, the serializer will emit code below:
`
var x;
(function () {
  var _1 = this;

  var _$0 = Object.defineProperty;
  var _0 = _1;

  _$0(_0, "x", {
    enumerable: true,
    configurable: false,
    writable: true,
    value: 1
  });

  _$0(_0, "x", {
    enumerable: true,
    configurable: false,
    writable: true,
    value: 2
  });

  _$0(_0, "x", {
    enumerable: true,
    configurable: false,
    writable: true,
    value: void 0
  });
}).call(this);
`
This is because the new logic will change the normal global assignment to use generator.emitDefineProperty() instead of generator.emitGlobalDeclaration(). 

I think we need some kind of logic to distinguish between global binding vs assignment and use generator.emitGlobalDeclaration() vs generator.emitGlobalAssignment() correspondingly.